calico_node/tests/st/calicoctl/test_default_pools.py

```diff
@@ -138,7 +138,11 @@ def test_default_pools(self, success_expected, param, value, exp_num_pools, ipip
                 "Didn't find ipip mode in pool %s" % pool
 
         # Check NAT setting
-        assert pool['spec']['nat-outgoing'] is True, "Didn't find nat enabled in pool %s" % pool
+        if param == "CALICO_IPV4POOL_CIDR":
```

Comment:
From the test error, it looks like you need to check if `nat-outgoing` is present in `pool['spec']`.
I think you could just switch the whole block here to something like:
```
if "nat-outgiong" in pool['spec']:
  assert pool['spec']['nat-outgoing'] is nat_outgoing, \
    "Wrong NAT default in pool %s, expected nat-outgoing to be %s" % (pool, nat_outgoing)
else:
  assert nat_outgoing is False, \
    "Wrong NAT default in pool %s, expecting nat-outgoing to be disabled" % pool
```

I don't think it is necessary to indicate v4 or v6.